osu.Desktop/Security/ElevatedPrivilegesChecker.cs

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Security.Principal;
+using osu.Framework;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Sprites;
+using osu.Game.Graphics;
+using osu.Game.Overlays;
+using osu.Game.Overlays.Notifications;
+
+namespace osu.Desktop.Security
+{
+    /// <summary>
+    /// Checks if the game is running with elevated privileges (as admin in Windows, root in Unix) and displays a warning notification if so.
+    /// </summary>
+    public class ElevatedPrivilegesChecker : Component
+    {
+        [Resolved]
+        protected NotificationOverlay Notifications { get; private set; }
+
+        private bool elevated;
+
+        [BackgroundDependencyLoader]
+        private void load()
+        {
+            elevated = isElevated();
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            if (!elevated)
+                return;
+
+            Notifications.Post(new ElevatedPrivilegesNotification());
+        }
+
+        private bool isElevated()
+        {
+            switch (RuntimeInfo.OS)
```

Comment:
To avoid any kind of weirdness, I'd surround this whole code with a `try-catch` and `return false;` in the `catch`. Never know what kind of exceptions may be thrown by platform code, and we don't want the game execution halting from a failure here.